src/components/decks/DeckListing.vue

```diff
@@ -46,28 +73,32 @@ export default {
     showMine: {
       type: Boolean,
       default: false,
+    },
+    showForUser: {
+      type: Number,
+      default: 0
     }
   },
   data: () => {
     return {
       isDisabled: false,
       filterText: '',
       phoenixborn: null,
+      shouldShowLegacyDecks: false,
       offset: 0,
       // This is the list of decks currently shown
       decks: null,
       deckCount: 0,
+      screenWidth: screen.width
     }
   },
   components: {
     DeckTable,
     ClearableSearch,
     PhoenixbornPicker,
+    FormatPicker
   },
   computed: {
-    showLegacy () {
-      return !!this.$route.meta.showLegacy
```

Comment:
This logic is actually still necessary. This is kind of a tricky one, because most of the other pages that use a deck listing are located at a completely different URL with the legacy status conferred through the route meta. See for instance the route defined here: https://github.com/onecrayon/ashes.live/blob/feature/decks/src/router.js#L63

To support both, we'll need:

* This component should still support showLegacy on the route meta, falling back to the query string if it doesn't exist
* For the `/decks/` routes we'll need a new meta entry with the URL for the legacy/non-legacy variant. If that exists, then the button in this component needs to use a `$router.push` to navigate instead of changing the query string.
* When adjusting the query string in this component, we should only do it if none of the aforementioned router meta are defined